game-core/src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -652,6 +652,23 @@ private Matches() {}
     };
   }
 
+  /** Checks if the unit type can be hit with AA fire by one of the firingUnits */
+  private static Predicate<UnitType> unitTypeCanByHitByAaFire(
+      final Collection<UnitType> firingUnits, final GameData gameData, final int battleRound) {
```

Comment:
WDYT to pass the data from `GameData` to this method rather than passing `GameData` to it?

`GameData` has a problem where it's passed everywhere and used by everything. That makes decoupling quite difficult, for example, testing involving with a game data almost always means doing a full blown integration test and loading an XML (which is extremely limiting as we may want to test perhaps 12 permutations of one element of game data for just one set of tests).

Anyways, the less dependencies we can have on game data for now-  the better.

Reply:
That would require changing `UnitAttachment#getTargetsAa` since this method just passes `gameData` to that method.  So for this PR, I don't think that is ok.